internal/cli/certificate.go

```diff
@@ -0,0 +1,403 @@
+		fmt.Printf("%s: %s\n", status, name)
+		if statuscode != http.StatusOK && statuscode != http.StatusCreated {
+			return fmt.Errorf("%s: applying CertificateSigningRequest: %s", name, string(response.Body))
+		}
+	}
+	return nil
+}
+
+func getCsr(o *CertificateOptions, name string, c *apiclient.ClientWithResponses, ctx context.Context) (*api.CertificateSigningRequest, error) {
+	response, err := c.ReadCertificateSigningRequestWithResponse(ctx, name)
+	if err != nil {
+		return nil, err
+	}
+
+	if response.HTTPResponse != nil {
+		statuscode := response.HTTPResponse.StatusCode
+		if statuscode != http.StatusOK && statuscode != http.StatusCreated {
+			return nil, fmt.Errorf("%s: reading CertificateSigningRequest: %s", name, string(response.Body))
+		}
+	}
+
+	currentCsr := api.CertificateSigningRequest{}
+	if err := json.Unmarshal([]byte(string(response.Body)), &currentCsr); err != nil {
+		return nil, err
+	}
+
+	return &currentCsr, nil
+}
+
+func createEmbeddedConfig(name string, c *apiclient.ClientWithResponses, ctx context.Context, priv crypto.PrivateKey, response *apiclient.EnrollmentConfigResponse) error {
+	pemPriv, err := fccrypto.PEMEncodeKey(priv)
+	if err != nil {
+		return err
+	}
+	response.JSON200.EnrollmentService.Authentication.ClientKeyData = base64.StdEncoding.EncodeToString(pemPriv)
+
+	marshalled, err := yaml.Marshal(response.JSON200)
+	if err != nil {
+		return fmt.Errorf("marshalling resource: %w", err)
+	}
+
+	_, err = fmt.Print(string(marshalled))
+	if err != nil {
+		return err
+	}
+
+	return nil
+}
+
+func writeCertFile(data []byte, path string) error {
+	fmt.Printf("writing data to %s...\n", path)
+	f, err := os.Create(path)
+	if err != nil {
+		return fmt.Errorf("creating file %s\n", path)
+	}
+	defer f.Close()
+	_, err = f.Write(data)
+	if err != nil {
+		return fmt.Errorf("writing data to %s: %w\n", path, err)
+	}
+
+	return nil
+}
+
+func createReferenceConfig(name string, currentCsr *api.CertificateSigningRequest, priv crypto.PrivateKey, response *apiclient.EnrollmentConfigResponse, path string) error {
+	if _, err := os.Stat(path); os.IsNotExist(err) {
+		err := os.MkdirAll(path, 0644)
+		if err != nil {
+			return fmt.Errorf("creating directory %s: %w\n", path, err)
+		}
+	}
+
+	// write out agent cert
+	certpath := path + name + ".crt"
+	cert := currentCsr.Status.Certificate
+	err := writeCertFile(*cert, certpath)
+	if err != nil {
+		return err
+	}
+
+	// write out ca cert
+	capath := path + name + "-ca-data.crt"
+	cadata, err := base64.StdEncoding.DecodeString(response.JSON200.EnrollmentService.Service.CertificateAuthorityData)
+	if err != nil {
+		return fmt.Errorf("decoding CA data: %w", err)
+	}
+	err = writeCertFile(cadata, capath)
+	if err != nil {
+		return err
+	}
+
+	// print config to stdout
+	response.JSON200.EnrollmentService.Authentication.ClientKeyData = path + name + ".key"
+	response.JSON200.EnrollmentService.Authentication.ClientCertificateData = certpath
```

Comment:
This must be the path where the _agent_ will find the files _on the device_ (/etc/flightctl/certs).

Reply:
I've done the following:
If the output type is `reference`, the key, cert, and ca-bundle files are emitted to the user-chosen directory. However, the agent config references `/etc/flightctl/certs/` + the filename.